File: src/com/mishiranu/instantimage/widget/SquareImageView.java
package com.mishiranu.instantimage.widget;

import android.content.Context;
import android.widget.ImageView;

public class SquareImageView extends ImageView
{
	public SquareImageView(Context context)
	{
		super(context);
	}
	
	@Override
	protected void onMeasure(int widthMeasureSpec, int heightMeasureSpec)
	{
		super.onMeasure(widthMeasureSpec, heightMeasureSpec);
		setMeasuredDimension(getMeasuredWidth(), getMeasuredWidth());
	}
}

File: src/com/mishiranu/instantimage/widget/SquareImageView.scala
package com.mishiranu.instantimage.widget

import android.content.Context
import android.widget.ImageView

class SquareImageView(context: Context) extends ImageView(context) {
  override protected def onMeasure(widthMeasureSpec: Int, heightMeasureSpec: Int): Unit = {
    super.onMeasure(widthMeasureSpec, heightMeasureSpec)
    setMeasuredDimension(getMeasuredWidth, getMeasuredWidth)
  }
}
